l3build-aux.lua

```diff
@@ -57,69 +75,89 @@ function setepoch()
+---@param target string
+---@param opts table
+---@return number 0 on proper termination, a non 0 error code otherwise.
+---@see many places, including latex2e/build.lua
+---@usage Public
+function call(modules, target, opts)
+  -- Turn the option table into a CLI option string
+  opts = opts or options
+  local cli_opts = ""
   for k,v in pairs(opts) do
-    if k ~= "names" and k ~= "target" then -- Special cases
-      local t = option_list[k] or { }
-      local arg = ""
+    if k ~= "names" and k ~= "target" then -- Special cases, to be removed!!!
+      local t = option_list[k] or {}
+      local value = ""
       if t["type"] == "string" then
-        arg = arg .. "=" .. v
-      end
-      if t["type"] == "table" then
+        value = value .. "=" .. v
+      elseif t["type"] == "table" then
         for _,a in pairs(v) do
-          if arg == "" then
-            arg = "=" .. a -- Add the initial "=" here
+          if value == "" then
+            value = "=" .. a -- Add the initial "=" here
           else
-            arg = arg .. "," .. a
+            value = value .. "," .. a
           end
         end
       end
-      s = s .. " --" .. k .. arg
+      cli_opts = cli_opts .. " --" .. k .. value
     end
   end
-  if opts["names"] then
-    for _,v in pairs(opts["names"]) do
-      s = s .. " " .. v
+  if opts.names then
+    for _, name in pairs(opts.names) do
+      cli_opts = cli_opts .. " " .. name
     end
   end
-  local scriptname = getscriptname()
-  for _,i in ipairs(dirs) do
-    local text = " for module " .. i
-    if i == "." and opts["config"] then
+  local script_name = get_script_name()
+  for _, module in ipairs(modules) do
+    local text
+    if module == "." and opts["config"] and #opts["config"] then
       text = " with configuration " .. opts["config"][1]
+    else
+      text = " for module " .. module
     end
     print("Running l3build with target \"" .. target .. "\"" .. text )
-    local errorlevel = run(
-      i,
-      "texlua " .. scriptname .. " " .. target .. s
+    local error_level = run(
+      module,
+      "texlua " .. script_name .. " " .. target .. cli_opts
     )
-    if errorlevel ~= 0 then
-      return errorlevel
+    if error_level ~= 0 then
+      return error_level
     end
   end
   return 0
 end
 
--- Unpack files needed to support testing/typesetting/unpacking
-function depinstall(deps)
-  local errorlevel
-  for _,i in ipairs(deps) do
-    print("Installing dependency: " .. i)
-    errorlevel = run(i, "texlua " .. getscriptname() .. " unpack -q")
-    if errorlevel ~= 0 then
-      return errorlevel
+---Unpack the given dependencies.
+---A dependency is the path of a directory relative to the main one.
+---@param deps table regular array of dependencies.
+---@return number 0 on proper termination, a non 0 error code otherwise.
+---@see stdmain, check, unpack, typesetting
+---@usage Private?
+function dep_install(deps)
+  local error_level
```

Comment:
I guess other renames are a matter of 'style' (I favoured single runs of `a-z` when I wrote `l3build`, others have added `_`), but for `errorlevel`, there is a long-established convention that it's one word.

Reply:
This is a matter of both style and efficiency. `errorlevel` smells a bit vintage TeX isn't it? Nowadays we have `\NewDocumentCommand` and `\l_tmpa_str` ;-)
BTW, is it told somewhere that the error level must be nonnegative?

Reply:
All I mean is that `errorlevel` is a very widely used name (e.g. Windows batch files), so I'd favour sticking to it.

In terms of possible return values, I have 'fun' with tools we call setting odd numbers, I think due to twos-complement somewhere. So the only thing I would rely on is `0` is 'success' and non-zero is 'failure'.

Reply:
The fact that `errorlevel` is already known in a widely used different context and a different usage also justifies to use a different name here. 

This is not a 0/non-zero alternative, it is a 0/positive alternative. clean, typeset and install both cumulate the error levels of various commands. If exactly one command returned 1 whereas exactly another returned -1, then the sum would be 0 and the result would be a falsy success.

Reply:
> The fact that `errorlevel` is already known in a widely used different context and a different usage also justifies to use a different name here.
> 
> This is not a 0/non-zero alternative, it is a 0/positive alternative. clean, typeset and install both cumulate the error levels of various commands. If exactly one command returned 1 whereas exactly another returned -1, then the sum would be 0 and the result would be a falsy success.

To-date, we've not had a case where that's an issue. I'd suggest tighening up where values return from an external process: we can do something like `if os.execute("thing") > 0 then errorlevel = errorlevel + 1` or similar. To be honest, we could also just fail at the first dodgy return: in the end, that's likely better style.

Reply:
So `if os.execute("thing") then return 1 end` for all `thing` might be best.